src/rules/require-baseline.js

```diff
@@ -0,0 +1,564 @@
+	 */
+	hasPropertyFunction(property, func) {
+		return this.#rules.some(rule => rule.hasFunction(property, func));
+	}
+
+	/**
+	 * Determines if any rule supports any functions.
+	 * @param {string} property The name of the property.
+	 * @returns {boolean} `true` if any rule supports the functions, `false` if not.
+	 */
+	hasPropertyFunctions(property) {
+		return this.#rules.some(rule => rule.hasFunctions(property));
+	}
+}
+
+//-----------------------------------------------------------------------------
+// Rule Definition
+//-----------------------------------------------------------------------------
+
+export default {
+	meta: {
+		type: /** @type {const} */ ("problem"),
+
+		docs: {
+			description: "Enforce the use of baseline features",
+			recommended: true,
+		},
+
+		schema: [
+			{
+				type: "object",
+				properties: {
+					available: {
+						enum: ["widely", "newly"],
+					},
+				},
+				additionalProperties: false,
+			},
+		],
+
+		defaultOptions: [
+			{
+				available: "widely",
+			},
+		],
+
+		messages: {
+			notBaselineProperty:
+				"Property '{{property}}' is not a {{availability}} available baseline feature.",
+			notBaselinePropertyValue:
+				"Value '{{value}}' of property '{{property}}' is not a {{availability}} available baseline feature.",
+			notBaselineAtRule:
+				"At-rule '@{{atRule}}' is not a {{availability}} available baseline feature.",
+			notBaselineType:
+				"Type '{{type}}' is not a {{availability}} available baseline feature.",
+		},
+	},
+
+	create(context) {
+		const availability = context.options[0].available;
+		const baselineLevel =
+			availability === "widely" ? BASELINE_HIGH : BASELINE_LOW;
+		const supportsRules = new SupportsRules();
+
+		/**
+		 * Checks a property value identifier to see if it's a baseline feature.
+		 * @param {string} property The name of the property.
+		 * @param {import("css-tree").Identifier} child The node to check.
+		 * @returns {void}
+		 */
+		function checkPropertyValueIdentifier(property, child) {
+			// named colors are always valid
+			if (namedColors.has(child.name)) {
+				return;
+			}
+			const possiblePropertyValues = propertyValues.get(property);
+
+			// if we don't know of any possible property values, just skip it
+			if (!possiblePropertyValues) {
+				return;
+			}
+
+			const propertyValueLevel = possiblePropertyValues.get(child.name);
+
+			if (propertyValueLevel < baselineLevel) {
+				context.report({
+					loc: child.loc,
+					messageId: "notBaselinePropertyValue",
+					data: {
+						property,
+						value: child.name,
+						availability,
+					},
+				});
+			}
+		}
+
+		/**
+		 * Checks a property value function to see if it's a baseline feature.
+		 * @param {import("css-tree").FunctionNodePlain} child The node to check.
+		 * @returns {void}
+		 **/
+		function checkPropertyValueFunction(child) {
+			const propertyValueLevel = types.get(child.name);
+
+			if (propertyValueLevel < baselineLevel) {
```

Comment:
Similar to https://github.com/eslint/css/pull/33#discussion_r1949169636, I think it wold be good to check if `types` has `child.name`.